components/automate-ui/src/app/modules/infra-proxy/create-environment-modal/create-environment-modal.component.ts

```diff
@@ -0,0 +1,336 @@
+  public showConstraint = false;
+
+  public firstFormGroup: FormGroup;
+  public thirdFormGroup: FormGroup;
+  public fourthFormGroup: FormGroup;
+  public constraintKeysp: string[] = [];
+  public cookbooks: Cookbook[] = [];
+  public default_attr_value = '{}';
+  public override_attr_value = '{}';
+  public server: string;
+  public org: string;
+  public per_page = 9;
+  public page = 1;
+  public name_idp = '';
+  public jsonString: string;
+  public dattrParseError: boolean;
+  public oattrParseError: boolean;
+  public data: any;
+  public conflictErrorEvent = new EventEmitter<boolean>();
+  public close = new EventEmitter();
+  public constraintArray: Array<DynamicGrid> = [];
+  public items: Environment[] = [];
+  public textareaID: string;
+  public dynamicArrayp: Array<DynamicGrid> = [];
+  private isDestroyed = new Subject<boolean>();
+
+  constructor(
+    private fb: FormBuilder,
+    private store: Store<NgrxStateAtom>,
+    private telemetryService: TelemetryService
+
+  ) {
+
+    this.firstFormGroup = this.fb.group({
+      name: ['', [Validators.required, Validators.pattern(Regex.patterns.NON_BLANK)]],
+      description: ['', [Validators.required, Validators.pattern(Regex.patterns.NON_BLANK)]]
+    });
+
+    this.thirdFormGroup = this.fb.group({
+      dattr: ['']
+    });
+
+    this.fourthFormGroup = this.fb.group({
+      oattr: ['', [Validators.required]]
+    });
+
+  }
+
+  ngOnInit(): void {
+    this.openEvent.pipe(takeUntil(this.isDestroyed))
+      .subscribe(() => {
+      this.conflictError = false;
+      this.visible = true;
+      this.items = this.environmentsList;
+      this.showdrag = true;
+      this.showConstraint = true;
+      this.server = this.serverId;
+      this.org = this.orgId;
+    });
+
+    const payload = {
+      environmentName: '',
+      page: this.page,
+      per_page: this.per_page,
+      server_id: this.serverId,
+      org_id: this.orgId
+    };
+
+    this.loadCookbookConstraint();
+
+    this.store.select(saveStatus)
+    .pipe(
+      takeUntil(this.isDestroyed),
+      filter(state => state === EntityStatus.loadingSuccess))
+      .subscribe(state => {
+        this.creating = false;
+        if (state === EntityStatus.loadingSuccess) {
+          this.store.dispatch(new GetEnvironments(payload));
+          this.closeCreateModal();
+        }
+      });
+
+    combineLatest([
+      this.store.select(saveStatus),
+      this.store.select(saveError)
+    ]).pipe(
+      takeUntil(this.isDestroyed),
+      filter(([state, error]) => state === EntityStatus.loadingFailure && !isNil(error)))
+      .subscribe(([_, error]) => {
+        if (error.status === HttpStatus.CONFLICT) {
+          this.conflictErrorEvent.emit(true);
+          this.conflictError = true;
+        } else {
+          this.store.dispatch(new GetEnvironments(payload));
+          this.creating = false;
+
+          // Close the modal on any error other than conflict and display in banner.
```

Comment:
More clear to say, e.g.:
```suggestion
          // Close the modal on any other error because it will be displayed in the banner.
```

Reply:
Updated the comment.